packages/slice-machine/src/modules/selectedSlice/reducer.ts

```diff
@@ -0,0 +1,273 @@
+        action.payload.mockConfig,
+        action.payload.variationId,
+        action.payload.widgetArea,
+        action.payload.newKey
+      );
+
+      const newModel = { ...prevState.Model, mockConfig: updatedConfig };
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+      };
+    }
+    case getType(generateSliceScreenshotCreator): {
+      if (!prevState) return prevState;
+
+      const newModel = {
+        ...prevState.Model,
+        screenshotUrls: action.payload.screenshots,
+      };
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+      };
+    }
+    case getType(generateSliceCustomScreenshotCreator): {
+      if (!prevState) return prevState;
+      const { variationId, screenshot } = action.payload;
+      const { Model: prevModel } = prevState;
+
+      const screenshots = prevModel.variations.reduce((acc, variation) => {
+        if (variation.id === variationId) {
+          return {
+            ...acc,
+            [variationId]: screenshot,
+          };
+        }
+        if (prevModel.screenshotUrls?.[variation.id]) {
+          return {
+            ...acc,
+            [variation.id]: prevModel.screenshotUrls?.[variation.id],
+          };
+        }
+        return acc;
+      }, {});
+
+      const newModel = { ...prevModel, screenshotUrls: screenshots };
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+      };
+    }
+    case getType(saveSliceCreator): {
+      if (!prevState) return prevState;
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(action.payload.state),
+      };
+    }
+    case getType(pushSliceCreator): {
+      if (!prevState) return prevState;
+      const newModel = {
+        ...prevState.Model,
+        __status: LibStatus.Synced,
+        initialScreenshotUrls: prevState.Model.screenshotUrls,
+        remoteVariations: prevState.Model.variations,
+      };
+
+      return { ...prevState, Model: updateTouchedAndStatus(newModel) };
+    }
+    case getType(copyVariationSliceCreator): {
+      if (!prevState) return prevState;
+      const { key, name, copied } = action.payload;
+      const newVariation = Variation.copyValue(copied, key, name);
+      const newModel = {
+        ...prevState.Model,
+        variations: prevState.Model.variations.concat([newVariation]),
+      };
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+        variation: newVariation,
+      };
+    }
+    default:
+      return prevState;
+  }
+};
+
+const updateTouchedAndStatus = (model: SliceState) => {
+  const isTouched =
+    !equal(model.initialVariations, model.variations) ||
+    !equal(model.initialMockConfig, model.mockConfig);
+  let status = model.__status;
+
+  const isScreenshotModified = !equal(
+    model.screenshotUrls,
+    model.initialScreenshotUrls
+  );
+
+  // Should be the slice be pushed ?
```

Comment:
Pushed? 

When it's touched i think that means it can be pushed, we really should think about the naming schema here as it could mean a few things

Reply:
Yhink that's correct. We are checking whether our local variations match the remote ones, if not it means we have some new ones that need to be pushed. Changed the comment to be a bit more explicit and not contain info on how the variable is used, simply to specify what it means

Reply:
nice